code/game/objects/items/devices/scanners.dm

```diff
@@ -274,109 +226,106 @@ GENE SCANNER
 
-	//Organ damages report
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		var/minor_damage
-		var/major_damage
-		var/max_damage
-		var/report_organs = FALSE
-
-		//Piece together the lists to be reported
-		for(var/O in H.internal_organs)
-			var/obj/item/organ/organ = O
-			if(organ.organ_flags & ORGAN_FAILING)
-				report_organs = TRUE	//if we report one organ, we report all organs, even if the lists are empty, just for consistency
-				if(max_damage)
-					max_damage += ", "	//prelude the organ if we've already reported an organ
-					max_damage += organ.name	//this just slaps the organ name into the string of text
-				else
-					max_damage = "\t<span class='alert'>Non-Functional Organs: "	//our initial statement
-					max_damage += organ.name
-			else if(organ.damage > organ.high_threshold)
-				report_organs = TRUE
-				if(major_damage)
-					major_damage += ", "
-					major_damage += organ.name
-				else
-					major_damage = "\t<span class='info'>Severely Damaged Organs: </span>"
-					major_damage += organ.name
-			else if(organ.damage > organ.low_threshold)
-				report_organs = TRUE
-				if(minor_damage)
-					minor_damage += ", "
-					minor_damage += organ.name
-				else
-					minor_damage = "\t<span class='info'>Mildly Damaged Organs: </span>"
-					minor_damage += organ.name
-
-		if(report_organs)	//we either finish the list, or set it to be empty if no organs were reported in that category
-			if(!max_damage)
-				max_damage = "\t<span class='alert'>Non-Functional Organs: </span>"
-			else
-				max_damage += "</span>"
-			if(!major_damage)
-				major_damage = "\t<span class='info'>Severely Damaged Organs: </span>"
-			else
-				major_damage += "</span>"
-			if(!minor_damage)
-				minor_damage = "\t<span class='info'>Mildly Damaged Organs: </span>"
-			else
-				minor_damage += "</span>"
-			to_chat(user, minor_damage)
-			to_chat(user, major_damage)
-			to_chat(user, max_damage)
+
+		// Organ damage
+		if (H.internal_organs && H.internal_organs.len)
+			var/render = FALSE
+			var/toReport = "<span class='info ml-1'>Organs:</span>\
+				<table class='ml-2'><tr>\
+				<td style='width:6em;'><font color='#0000CC'><b>Organ</b></font></td>\
+				[advanced ? "<td style='width:3em;'><font color='#0000CC'><b>Dmg</b></font></td>" : ""]\
+				<td style='width:12em;'><font color='#0000CC'><b>Status</b></font></td>"
+
+			for(var/obj/item/organ/organ in H.internal_organs)
+				var/status = ""
+				if (organ.organ_flags & ORGAN_FAILING) status = "<font color='#E42426'>Non-Functional</font>"
+				else if (organ.damage > organ.high_threshold) status = "<font color='#EC6224'>Severely Damaged</font>"
+				else if (organ.damage > organ.low_threshold) status = "<font color='#F28F1F'>Mildly Damaged</font>"
+				if (status != "")
+					render = TRUE
+					toReport += "<tr><td><font color='#0000CC'>[organ.name]</font></td>\

```

Comment:
you should really be creating a list and .joining it all together at the end.

Also, I dislike this trend of shoving html formatting into to_chat and the goonchat window. 

There's no telling what it looks like in the fallback window.

This kind of stuff should go into browse windows

Reply:
we have lists now welcome to the render pipeline zone

margin-left probably ok for fallback because its just offsetting from the left but tables might be funky